packages/fxa-payments-server/src/lib/errors.ts

```diff
@@ -31,6 +44,10 @@ let errorMessageIndex: { [key: string]: string } = {
   coupon_expired: 'coupon-expired-error',
   card_error: 'card-error',
   // todo: handle "parameters_exclusive": "Your already subscribed to _product_"
+  // Currency errors
+  'Funding source country does not match plan currency.':
+    'country-currency-mismatch',
+  'Changing currencies is not permitted.': 'currency-currency-mismatch',
```

Comment:
Ah I finally figured out where these keys are coming from -- the auth server. It's a little odd having these new keys break the pattern, though I guess the other ones are almost all Stripe error codes.

Reply:
Yes. And there may be paypal specific codes in the future. I agree this whole module feels janky, but I wasn't going to tackle refactoring it.